### PR TITLE
Icon packs

### PR DESCRIPTION
**Pull request by sglanzer-fire**
Adds the icon pack #feature# to the build process and the `frost-icon` component.  Apologies for the messy duplication in `index.js`, I didn't have time to clean up all of the code.
# CHANGELOG

Added the icon pack feature to `frost-icon`, see the documentation at http://ciena-frost.github.io/ember-frost-core/#/icons for details

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -46,34 +47,119 @@ module.exports = {
...
+      } else if (fileName.endsWith('.svg')) {
+        iconNames.push(`${subDir}/${fileName.substr(0, fileName.length - 4)}`)
+      }
+    })
+
+    return iconNames
+  },
+  
+  // Present purely to allow programmatic access to the icon packs and icon names (for demo purposes)
   treeForAddon: function (tree) {
```
Comment (sglanzer-fire) [inline index.js:63]: There is duplication in the detection of icon packs between `treeForAddon` and `treeForPublic` that I didn't have time to clean up (sorry) - be nice ;)

**Comment by sglanzer-fire** [comment]
The documentation that will be hosted off the changelog link is in the frost-icons.md

File: index.js [hunk 2]
```diff
@@ -46,34 +46,111 @@ module.exports = {
...
+        }
+      })
+      .map((addonPackage) => {
+        const iconPack = addonPackage.pkg['ember-frost-icon-pack']
+        const iconPackPath = iconPack.path || 'svgs'
+        const addonIconPackPath = path.join(addonPackage.path, iconPackPath)
+
+        var svgFunnel
+        if (iconPack.name === 'frost' && isLegacy) {
+          console.log('hi')
```
Comment (rox163) [inline index.js]: missed this in clean up?

File: addon/components/frost-icon.js [hunk 3]
```diff
@@ -1,22 +1,34 @@
...
+    pack: PropTypes.string,
+    icon: PropTypes.string.isRequired
+  },
+  tagName: 'svg',
 
-  // TODO This is a pretty nasty way to inline svgs - let's look into a better way
-  svg: Ember.computed('icon', 'class', function () {
-    let svg = Ember.get(svgs, this.get('icon').replace(/\//g, '.'))
+  didReceiveAttrs ({newAttrs}) {
+    deprecate(
```
Comment (rox163) [inline addon/components/frost-icon.js:18]: :+1:

File: index.js (hunk 2 above)
Comment (sglanzer-fire) [inline index.js]: Just being friendly :)

File: addon/components/frost-icon.js [hunk 4]
```diff
@@ -1,22 +1,34 @@
 import Ember from 'ember'
-import _ from 'lodash/lodash'
-import svgs from 'ember-frost-core/svgs'
+import { PropTypes } from 'ember-prop-types'
 import layout from '../templates/components/frost-icon'
+import _ from 'lodash'
+
+const { deprecate } = Ember
 
 export default Ember.Component.extend({
```
Comment (sandersky) [inline addon/components/frost-icon.js]:
``` js
const {Component, deprecate} = Ember

export default Component.extend({
  …
})
```

File: addon/components/frost-icon.js (hunk 3 above)
Comment (sandersky) [inline addon/components/frost-icon.js:18]: Didn't know Ember provided a deprecation method, nice!

File: tests/dummy/app/pods/icons/controller.js [hunk 5]
```diff
@@ -1,7 +1,9 @@
 import Ember from 'ember'
-import svgs from 'ember-frost-core/svgs'
+import iconPacks from 'ember-frost-core/icon-packs'
 import _ from 'lodash/lodash'
 
+const { computed } = Ember
+
 export default Ember.Controller.extend({
```
Comment (sandersky) [inline tests/dummy/app/pods/icons/controller.js]:
``` js
const {computed, Controller} = Ember 

export default Controller({
  …
})
```

**Comment by sandersky** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-core/pull-request/96/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-core)

**Pull request merged**